analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -29,5 +29,80 @@ public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute()
+    {
+        return IsAccessibleFromOtherAssemblies(Node)
+               && (IsSupportedMethod(Node) || IsSupportedPropertyAccessor(Node));
+
+        static bool IsSupportedMethod(SyntaxNode node) =>
+            node is BaseMethodDeclarationSyntax { ParameterList.Parameters.Count: > 0 } method
+            && MethodDereferencesArguments(method);
+
+        static bool IsSupportedPropertyAccessor(SyntaxNode node) =>
+            node is AccessorDeclarationSyntax { RawKind: (int)SyntaxKind.SetAccessorDeclaration or (int)SyntaxKindEx.InitAccessorDeclaration } accessor
+            && IsPropertyAccessorAccessibleFromOtherAssemblies(accessor.Modifiers);
+
+        static bool IsAccessibleFromOtherAssemblies(SyntaxNode node) =>
+            node.AncestorsAndSelf().OfType<MemberDeclarationSyntax>().FirstOrDefault() is { } containingMember
+            && node.Ancestors().OfType<BaseTypeDeclarationSyntax>().FirstOrDefault() is { } containingType
+            && IsMemberAccessibleFromOtherAssemblies(containingMember.Modifiers(), containingType)
+            && IsTypeAccessibleFromOtherAssemblies(containingType.Modifiers);
+
+        static bool IsMemberAccessibleFromOtherAssemblies(SyntaxTokenList modifiers, BaseTypeDeclarationSyntax containingType) =>
+            IsPublicOrProtectedOrProtectedInternal(modifiers)
+            || (containingType is InterfaceDeclarationSyntax && HasNoDeclaredAccessabilityModifier(modifiers));
```

Comment:
I think you can drop the `HasNoDeclaredAccessabilityModifier` since interface members cannot have any access modifiers.
```suggestion
            || containingType is InterfaceDeclarationSyntax;
```

Reply:
Since C# 8 [interface methods can have all access modifiers](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods#modifiers-in-interfaces).

Reply:
But this means you must also check for `public`, `protected`, and `protected internal`, no?

Reply:
Yes, that's exactly what it does.